app/view/twig/_nav/_secondary-content.twig

```diff
@@ -7,18 +7,38 @@
     {% for name, menu in content_menus.main.children %}
         {% set active = (page_nav == 'Content/*' and context.contenttype.slug|default == name) %}
         {% set sub = [] %}
+        {% set singleton = false %}
+
+        {% if app.config.get('contenttypes')[name].singleton %}
+            {% setcontent records = name limit 1 nohydrate %}
+            {% set path = '' %}
+            {% set has_records = false %}
+
+            {%  if records|length == 1 %}
+                {% set path =  path('editcontent', {contenttypeslug: name, id: records|first.id}) %}
+                {#{% set has_records = true %}#}
+            {% else %}
+                {% set path = menu.get('new') %}
+            {% endif %}
+
+            {% set singleton = { 'uri': path, 'icon': menu.icon, 'label': menu.label } %}
+        {% endif %}
+
```

Comment:
As above, this entire file's logic is better handled in PHP … `Bolt\Menu\MenuEntry` & `Bolt\Menu\Builder\AdminContent` might give you more hints